### ISSUE TITLE
HealthIndicator always returning UNKNOWN status even if connected

### ISSUE THREAD

**Issue by mborgraeve** [spring-cloud/spring-cloud-stream-binder-kafka#1006]
Hey there !

I'm experiencing an issue with Spring Cloud Stream, the Kafka Stream Binder, and the auto configured Health Indicator for the Actuator.

[Here is a sample minimal project](https://github.com/mborgraeve/kafka-stream-health-indicator-sample) that reproduces my issue. You'll need a Kafka running (I'm working with an older 2.2.0 version, which constraints me to keep the SR8 and avoid SR9 and further) with two topics, `IN` and `OUT`.
Just startup the application (after adapting the `broker`'s address/port of course), you may try to send it a message and see that it goes through without issue.

I'm then calling the Health endpoint, which returns me the following:
```json
{
  "status": "UP",
  "components": {
    "binders": {
      "status": "UNKNOWN",
      "components": {
        "enricherKStreamBinder": {
          "status": "UNKNOWN"
        }
      }
    },
    "diskSpace": {
      "status": "UP",
      "details": {
        "total": 499963174912,
        "free": 344081379328,
        "threshold": 10485760,
        "exists": true
      }
    },
    "ping": {
      "status": "UP"
    }
  }
}
```

As the application is working as properly and can talk to Kafka, I would have expected to see a `$.component.binders.status` and `$.component.binders.components.enricherKStreamBinder` set to `UP`.

When debugging, I see that the code goes through `org.springframework.cloud.stream.config.BindersHealthIndicatorAutoConfiguration.BindersHealthContributor#getContributor()`, called with an empty `Map` as argument, causing it to return the static `private static final HealthIndicator UNKNOWN = () -> {return Health.unknown().build();};`.
I naively would expect the map to not be empty there, but I'm no expert.

Hoping you can help, it may only be a configuration issue.
Thanks a lot for your work !

And of course feel free to redirect me if i'm in the wrong repo, or anything.

**Comment by mborgraeve** [comment]
Apologies, wrong repo...

**Comment by mborgraeve** [comment]
Apologies, mixed up the repos. This one seems to be the one, feel free to redirect me if not !

**Comment by sobychacko** [comment]
@mborgraeve Any chance you can upgrade to `3.0.x`? The version you are using `2.2.x` is based on Spring Boot 2.1.x which is [EOL'ed](https://spring.io/blog/2019/12/10/spring-boot-2-1-x-eol-november-1st-2020).

**Comment by mborgraeve** [comment]
No, unfortunately I have no control over the Kafka setup. I wish we could upgrade too...

**Comment by sobychacko** [comment]
@mborgraeve Sorry for the confusion. I didn't realize that the `2.2.x` you mentioned is the Kafka version. You are using the latest Spring Cloud Stream, so I will see if it is an issue. Which SR8 did you mention above?

**Comment by mborgraeve** [comment]
Ahh I see, no worries.
SR8 is the `spring-cloud-dependencies` train version, `Hoxton.SR8`, it's in the `pom.xml`

**Comment by sobychacko** [comment]
@mborgraeve Is this working if you don't use multi binder setup? Can you remove the multi binder configuration and make it a single binder context application and see if that works? I am still trying to see if there are some issues with multi binder setup and health check in Kafka Streams binder. Thanks!

**Comment by sobychacko** [comment]
@mborgraeve There was an issue in the way health indicators were configured when using an application with a multi-binder context. This is now addressed on both master and 3.0.x branches. Thanks for pointing out this issue.

**Comment by mborgraeve** [comment]
For some reason I wasn't notified of your answers, apologies.
Thanks a lot, and glad I could help.

> This is now addressed on both master and 3.0.x branches. 
I'm unsure how this fix would end up in the Spring Cloud Dependencies release train. I was using `spring-cloud-dependencies:Hoxton.SR8`, which pulls `soring-cloud-stream:Horsham.SR8`, which pulls `spring-cloud-stream-binder-kafka-streams:3.0.8`. Moving to `Hoxton.SR9` brings `Horsham.SR9` and `3.0.9`.

Would these be updated with your fix, or should I update to SR9 or another release train version?

**Comment by sobychacko** [comment]
@mborgraeve The fixes are not released yet. They will be part of the next Horsham release train (`Horsham.SR11`). We can try to include that in `Hoxton.SR10` of spring cloud. I will have to find out if/when a release for that is scheduled. 
There will be an SR1 for `3.1.x` line release (current master) coming up later this month which will be picked up by spring-cloud `2020.0.1` release, but that is using a different version of spring boot.
